### DISCUSSION TITLE
Vouchers/Discounts codes

### DISCUSSION THREAD

**Discussion by Bloodlex** [QuickShop-Community/QuickShop-Hikari#604] [Ideas]
Hey,

What do you think about introducing some kind of vouchers, discount codes or time discount for player shops? It would be awesome to give it for specific players as an event prize or just a seasonal discount for some shops for instance.

Me and my players would be very grateful for such a possibility if you also find it interesting 🙂.

**Comment by Ghost-chu** [comment]
I will definitely consider it, but I think it should be made as an independent extension plug-in.
In terms of the current QuickShop API, it is completely possible to implement this function. Unfortunately, it takes a lot of time, so Christmas may not be able to catch up.

**Reply by Bloodlex** [reply]
Hey, thanks for the answer, I'm looking forward to it! 🙂.

**Comment by Ghost-chu** [comment]
My idea is create a discount code feature.  

A discount code can apply to globally or for specific shop.  
* global discount - Player will still receive the full rewards from discount, basiclly is a server scope activity.
* personal discount - Owner and staffs are able to create a discount code for their shops, and player use the code to purchase specific shops will discount (of course, shop owner will receive more less rewards from purchase).

And a discount can be unlimited or limited of time, for example, x-mas discount can only be applied between 2022-12-24 to 2022-12-27 in real life time.

Also a discount should be countable, creator can set a limit to avoid abuse, for example, three times per player, or 5000 times in global.

**Comment by Ghost-chu** [comment]
The current plan is that players who need to use discount code will need to enter the command `/qs discount <code>` first, and discount code will be remembered until the server restarts and will be automatically applied to any subsequent purchases unless the player actively uses `/qs discount unset` to forget it.


Discount codes are automatically used when players purchase items in stores if they comply with the discount policy.

**Comment by Ghost-chu** [comment]
![](https://media.discordapp.net/attachments/942378854486528011/1053671753580286033/image.png?width=1302&height=767)
just some showcase.

**Comment by Ghost-chu** [comment]
![image](https://user-images.githubusercontent.com/30802565/208246083-500a7235-e6ea-411f-b669-14a3d45369c5.png)
more showcase

**Comment by Ghost-chu** [comment]
Public beta with 3.6.0.0-beta2 now.  
https://modrinth.com/plugin/quickshop-hikari/version/3.6.0.0-beta2

**Comment by Zxce3** [comment]
nice
